scheduling-manager/src/cucumber/java/cucumber/steps/scheduling/daemon/ReplaceAfterTerminateSteps.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package cucumber.steps.scheduling.daemon;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.amazonaws.blox.scheduling.scheduler.engine.EnvironmentDescription;
+import com.amazonaws.blox.scheduling.scheduler.engine.EnvironmentDescription.EnvironmentDescriptionBuilder;
+import com.amazonaws.blox.scheduling.scheduler.engine.Scheduler;
+import com.amazonaws.blox.scheduling.scheduler.engine.SchedulingAction;
+import com.amazonaws.blox.scheduling.scheduler.engine.StartTask;
+import com.amazonaws.blox.scheduling.scheduler.engine.daemon.ReplaceAfterTerminateScheduler;
+import com.amazonaws.blox.scheduling.state.ClusterSnapshot;
+import com.amazonaws.blox.scheduling.state.ClusterSnapshot.ContainerInstance;
+import com.amazonaws.blox.scheduling.state.ClusterSnapshot.Task;
+import cucumber.api.DataTable;
+import cucumber.api.java8.En;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ReplaceAfterTerminateSteps implements En {
+
+  private EnvironmentDescription environment;
+  private ClusterSnapshot snapshot;
+  private Scheduler scheduler = new ReplaceAfterTerminateScheduler();
+
+  private Set<SchedulingAction> actions = new HashSet<>();
+
+  public ReplaceAfterTerminateSteps() {
+    Given(
+        "^a Daemon environment named \"([^\"]*)\":$",
+        (String name, DataTable properties) -> {
+          EnvironmentDescriptionBuilder builder = environmentDescriptionFromTable(properties);
+
+          environment = builder.environmentName(name).build();
```

Comment:
shouldn't this be pulled from dataservice?

Reply:
It will, in the integration tests; these are unit tests.